src/base/static/js/utils.js

```diff
@@ -129,20 +110,33 @@ var self = (module.exports = {
     if (components.img.startsWith("data:")) {
       // If the image was just created and has a data url, fetch the attachment
       // collection to obtain the S3 url before contacting the sharing microservice.
-      model.attachmentCollection.fetch({
-        reset: true,
-        success: function(collection) {
-          components.img = collection.first().get("file")
-          var queryString = self.buildSharingQuerystring(components)
-          self.initiateShare(service, shareUrl, queryString)
-        },
+      return new Promise((resolve, reject) => {
+        model.attachmentCollection.fetch({
+          reset: true,
+          success: collection => {
+            components.img = collection.first().get("file")
+            const queryString = this.buildSharingQuerystring(components)
+            resolve(encodeURIComponent(`${shareUrl}${queryString}`))
+          },
+          error: _ => reject(_),
+        })
       })
     } else {
-      var queryString = this.buildSharingQuerystring(components)
-      this.initiateShare(service, shareUrl, queryString)
+      // return a promise that immediately resolves to our share url:
+      const queryString = this.buildSharingQuerystring(components)
+      return Promise.resolve(encodeURIComponent(`${shareUrl}${queryString}`))
     }
   },
 
+  onSocialShare: function(model, service) {
+    this.getSocialUrl(model).then(shareUrl => {
+      let url = service === 'facebook'
+              ? `https://www.facebook.com/sharer/sharer.php?u=${shareUrl}`
+              : `https://twitter.com/intent/tweet?url=${shareUrl}`
+      window.open(url, '_blank').focus()
```

Comment:
Maybe we want to open these in a popup? I think it's more usable than switching to a new tab.